### ISSUE TITLE
size mismatch in `config_ia2c_grid` PyTorch implementation 

### ISSUE THREAD

**Issue by Jackory** [cts198859/deeprl_network#12]
When I use the PyTorch implementation to train with config_ia2c_grid, I encounter this bug
![image](https://user-images.githubusercontent.com/52418786/110779163-2c068300-829e-11eb-94f9-72dc623cc7f5.png)
Could anyone give me some feedback? Thanks!

**Comment by cts198859** [comment]
The pytorch version is not fully tested. The root cause may be similar to that in https://github.com/cts198859/deeprl_network/issues/9#issue-709480785.

**Comment by Jackory** [comment]
Duplicate of #9. So what should I do to fix this bug, can you give some instructions in detail.

**Comment by cts198859** [comment]
Thanks for offering the fix! The error raised during the backward propagation, since the dim of sampled state from transition buffer does not match that assumed by the model. One fix may be selecting the first 36 elements of the state, i.e., in L92 of agents/policies.py: https://github.com/cts198859/deeprl_network/blob/b5ff5131b5ec97ad585ba3f54bc311e155312969/agents/policies.py#L92
        obs = torch.from_numpy(obs).float() => obs = torch.from_numpy(obs[:, :self.n_s]).float()

But the thing bothering me is that why there is no error in forward propagation. The state stored to the buffer should have the same dimension as that observed in forward propagation. This may need further investigation (e.g., comparing with the tf code)

https://github.com/cts198859/deeprl_network/blob/c279b1fc83ecb7367d38f4a8ada5f3cf585aec59/utils.py#L181

**Comment by Jackory** [comment]
Thanks for your quick reply!
